src/libs/app-icon.ts

```diff
@@ -0,0 +1,117 @@
+import path from "path";
+import * as asyncfile from "async-file";
+import * as download from "image-downloader";
+import * as Meta from "../Meta";
+import { Environment } from "../types/environment.types";
+import Jimp from "jimp";
+import { log } from "../output-channel";
+import { appsIconTempDir } from "../temp-dir";
+
+/**
+ * Downloads PNG icon from Make API and stores it in local temp dir.
+ * Also creates the inverted icon and (optionally) public icon.
+ * @returns Icon version. Is is used for icon filemame postfix.
+ *          0 = no icon.
+ */
+export async function downloadAndStoreAppIcon(
+	app: IApp, apiBaseUrl: string, apiAuthorization: string, environment: Environment, isAppOpensource: boolean
+): Promise<number> {
+	try {
+		let iconVersion = 0;  // TODO Investigate usage of iconVersion. Remove if not needed.
+		let dest: string;
+		do {
+			iconVersion++;
+			dest = path.join(appsIconTempDir, `${app.name}.${iconVersion}.png`);
+		} while (await asyncfile.exists(`${dest}.old`));  // TODO Investigate the usage of `.old` files. Remove if not needed.
+
+		if (!await asyncfile.exists(dest)) {
+			// Download new icon from API to localdir
+			try {
+				await download.image({
+					headers: {
+						"Authorization": apiAuthorization,
+						'x-imt-apps-sdk-version': Meta.version
+					},
+					url: (() => {
+						switch (environment.version) {
+							case 2:
+								return `${apiBaseUrl}/sdk/apps/${app.name}/${app.version}/icon/512`;
+							case 1:
+							default:
+								return `${apiBaseUrl}/app/${app.name}/${app.version}/icon/512`;
+						}
+					})(),
+					dest: dest
+				});
+			} catch (err: any) {
+				// App icon not saved in Make
+				return 0; // Version 0 = No icon
+			}
+		}
+
+		// Invert icon
+		try {
+			await invertPngAsync(dest);
+		} catch (err: any) {
+			log('error', `Failed to invert icon with URI ${dest}. Original error: ${err.message}`);
+		}
+
+		// Generate public icon
+		if (app.public && !isAppOpensource) {
```

Comment:
Here was the bug in previous implementation (code moved from AppsProvider), where Opensource apps icons was missing.

Root cause: It tried to use ".public.png" icon file for Opensource apps, but public file was not exist for opensource apps.